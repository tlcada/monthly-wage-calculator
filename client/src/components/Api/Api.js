import React from 'react'

import config from '../../../config/config'

const Api = () => (
  <p><a href={config.apiDoc} title='API documentation' target='_blank'>API documentation</a></p>
)

export default Api
