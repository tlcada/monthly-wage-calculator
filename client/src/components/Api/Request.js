/* global fetch:false */
import 'whatwg-fetch'

function Request (apiUrl) {
  return fetch(apiUrl, {
    method: 'GET',
    headers: {'Content-Type': 'application/json'}
  }).then(response => {
    return response.json()
  }).then(json => {
    return json
  }, err => {
    return err
  }).catch(function (err) {
    return Promise.reject(err)
  })
}

export { Request }
