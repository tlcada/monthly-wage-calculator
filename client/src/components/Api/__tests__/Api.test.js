import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Api from '../Api'

test('Api component should render as expected', () => {
  const component = shallow(<Api />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
