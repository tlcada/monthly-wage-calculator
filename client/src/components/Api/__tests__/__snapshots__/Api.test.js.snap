// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api component should render as expected 1`] = `
<p>
  <a
    href="about:blankapi"
    target="_blank"
    title="API documentation"
  >
    API documentation
  </a>
</p>
`;
